src/typechecker.erl

```diff
@@ -3980,14 +3952,31 @@ handle_type_error({type_error, mismatch, Ty, Expr}) ->
 handle_type_error(type_error) ->
     io:format("TYPE ERROR~n").
 
--spec print_type_error(string(),
-                       erl_parse:abstract_expr(),
+-spec describe_expr(erl_parse:abstract_expr()) -> string().
+describe_expr({atom, _, _})               -> "atom";
+describe_expr({bin, _, _})                -> "bit expression";
+describe_expr({char, _, _})               -> "character";
+describe_expr({cons, _, _, _})            -> "list";
+describe_expr({'fun', _, _})              -> "function expression";
```

Comment:
maybe this can be called "anonymous function" and `named_fun` "named function"?
(but you are right: "fun expression" is mentioned in the abstract format documentation)

Reply:
Fixed in follow up commit, forgot to fix before merge.